src/main/kotlin/io/github/yuutoproject/yuutobot/commands/Minigame.kt

```diff
@@ -0,0 +1,169 @@
+/*
+ * Open source bot built by and for the Camp Buddy Discord Fan Server.
+ *     Copyright (C) 2020  Kyuuto-devs
+ *
+ *     This program is free software: you can redistribute it and/or modify
+ *     it under the terms of the GNU General Public License as published by
+ *     the Free Software Foundation, either version 3 of the License, or
+ *     (at your option) any later version.
+ *
+ *     This program is distributed in the hope that it will be useful,
+ *     but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *     MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *     GNU General Public License for more details.
+ *
+ *     You should have received a copy of the GNU General Public License
+ *     along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+package io.github.yuutoproject.yuutobot.commands
+
+import io.github.yuutoproject.yuutobot.commands.base.AbstractCommand
+import io.github.yuutoproject.yuutobot.commands.base.CommandCategory
+import net.dv8tion.jda.api.EmbedBuilder
+import net.dv8tion.jda.api.JDA
+import net.dv8tion.jda.api.entities.Emote
+import net.dv8tion.jda.api.entities.Message
+import net.dv8tion.jda.api.entities.MessageEmbed
+import net.dv8tion.jda.api.entities.User
+import net.dv8tion.jda.api.events.message.guild.GuildMessageReceivedEvent
+import net.dv8tion.jda.api.events.message.guild.react.GuildMessageReactionAddEvent
+import net.dv8tion.jda.api.events.message.guild.react.GuildMessageReactionRemoveEvent
+import net.dv8tion.jda.api.hooks.ListenerAdapter
+import java.awt.Color
+
+enum class State {
+    OFF,
+    STARTING,
+    IN_PROGRESS
+}
+
+class MinigameListener(val minigame: Minigame) : ListenerAdapter() {
+    override fun onGuildMessageReceived(event: GuildMessageReceivedEvent) {
+        minigame.messageRecv(event)
+    }
+
+    override fun onGuildMessageReactionAdd(event: GuildMessageReactionAddEvent) {
+        minigame.reactionRecv(event)
+    }
+
+    override fun onGuildMessageReactionRemove(event: GuildMessageReactionRemoveEvent) {
+        minigame.reactionRetr(event)
+    }
+}
+
+class Minigame : AbstractCommand(
+    "minigame",
+    CommandCategory.GAME,
+    "Play a fun quiz with your friends!",
+    "Run `minigame` to begin a new game, and react within the countdown to join.\nRun `minigame skip` to skip a question you do not wish to answer."
+) {
+    var state = State.OFF
+    var players = mutableMapOf<User, Int>()
+    var countdown = 0
+    lateinit var startingMessage: Message
```

Comment:
This is unacceptable, there is a whole section in the troubleshooting guide as to why this is a bad idea https://github.com/DV8FromTheWorld/JDA/wiki/19%29-Troubleshooting#cannot-get-reference-as-it-has-already-been-garbage-collected